tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala

```diff
@@ -0,0 +1,137 @@
+package at.forsyte.apalache.io.lir
+
+import at.forsyte.apalache.tla.lir.oper.TlaOper.deinterleave
+import at.forsyte.apalache.tla.lir.oper.{TlaFunOper, TlaSetOper, TlcOper}
+import at.forsyte.apalache.tla.lir.values.{TlaBool, TlaInt, TlaStr}
+import at.forsyte.apalache.tla.lir.{OperEx, SeqT1, TlaConstDecl, TlaEx, TlaModule, TlaVarDecl, Typed, ValEx}
+
+import java.io.PrintWriter
+import java.util.Calendar
+import scala.collection.mutable
+
+/**
+ * This class produces counterexamples in the Informal Trace Format.
+ *
+ * @param writer a print writer to use
+ * @author Igor Konnov
+ */
+class ItfCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter {
+
+  /**
+   * The minimal value that can be reliably represented with Double in JavaScript.
+   */
+  val MIN_JS_INT: BigInt = -BigInt(2).pow(53) + 1
+
+  /**
+   * The maximal value that can be reliably represented with Double in JavaScript.
+   */
+  val MAX_JS_INT: BigInt = BigInt(2).pow(53) - 1
+
+  /**
+   * Produce a JSON representation of a counterexample in the ITF format
+   *
+   * @param rootModule the module that produced the counterexample
+   * @param states     a sequence of next states
+   * @return the JSON representation of the counterexample in the ITF format
+   */
+  def mkJson(rootModule: TlaModule, states: List[NextState]): ujson.Value = {
+    // merge constant initialization and variable initialization into a single state
+    val state0 = states match {
+      case Seq(constInit)                => constInit._2
+      case Seq(constInit, initState, _*) => constInit._2 ++ initState._2
+      case Seq()                         => throw new IllegalArgumentException("Expected at least one state, found none")
+    }
+    val mappedStates = state0 :: states.drop(2).map(_._2)
```

Comment:
If `states: List[NextState]`, you probably shouldn't be matching `Seq(...)`, but 
```suggestion
  def mkJson(rootModule: TlaModule, states: List[NextState]): ujson.Value = {
    // merge constant initialization and variable initialization into a single state
    val state0 = states match {
      case constInit :: Nil                => constInit._2
      case constInit :: initState :: _ => constInit._2 ++ initState._2
      case Nil                     => throw new IllegalArgumentException("Expected at least one state, found none")
    }
    val mappedStates = state0 :: states.drop(2).map(_._2)
```
You could also just do:
```suggestion
  def mkJson(rootModule: TlaModule, states: List[NextState]): ujson.Value = {
    // merge constant initialization and variable initialization into a single state
    if (states.isEmpty) throw new IllegalArgumentException("Expected at least one state, found none")
    val (prefix, suffix) = states.splitAt(2)
    val mappedStates = prefix.reduce( _ ++ _ ) :: suffix.map(_._2)
```